tools/prometheus/registry.go

```diff
@@ -29,3 +29,11 @@ func SafeExec(fn func()) {
 	}()
```

Comment:
I think the defer func is unnecessary.

Reply:
prevent trading failures due to prometheus panic